security-analysis/security-analysis-api/src/main/java/com/powsybl/security/SecurityAnalysisResultBuilder.java

```diff
@@ -57,18 +57,26 @@ private void addPostContingencyResult(PostContingencyResult result) {
      * @return a {@link PreContingencyResultBuilder} instance.
      */
     public PreContingencyResultBuilder preContingency() {
-        return new PreContingencyResultBuilder();
+        return new PreContingencyResultBuilder(context);
+    }
+
+    public PreContingencyResultBuilder preContingency(SecurityAnalysisResultContext context) {
```

Comment:
Maybe we should rename the argument to avoid confusion with the class field.

Reply:
```java
        /**
         * Adds a {@link LimitViolation} to the builder.
         * The default result context would be supplied to interceptors.
         * @param violation
         * @return
         */
        public B addViolation(LimitViolation violation) {
            addViolation(violation, resultContext);
            return (B) this;
        }
```
I am not sure for this part.

Reply:
It looks good to me :) As a convenience we allow the implementation to provide only a context at the beginning, and we pass it along to all calls.
It ensures that a context is always given to the interceptors.